src/c-writer.cc

```diff
@@ -3085,10 +3130,25 @@ void CWriter::Write(const ExprList& exprs) {
 
       case ExprType::Call: {
         const Var& var = cast<CallExpr>(&expr)->var;
+        assert(var.is_name());
+
         const Func& func = *module_->GetFunc(var);
         Index num_params = func.GetNumParams();
         Index num_results = func.GetNumResults();
+
+        if (isVarargsAllocatorFun(var.name())) {
+          assert(num_results == 1);
+          SetVarargs(num_params);
+          break;
+        }
+
+        if (IsVarargs()) {
+          num_params += TakeVarargs();
+          num_params -= 1;
```

Comment:
// Replacing last parameter with vararg parameters

Reply:
I tweaked the comments some more to explain things better.